app/screens/component_library/button.cl.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```

Comment:
Q: how come at MM, we don't name the file like we would the component?

ButtonComponentLibrary.tsx vs button.cl.tsx?

Reply:
In general we do. In this case I am making up a new standard just because :sweat_smile:  The idea here is that since it is the `component library` for the `button` component, then... `button.cl.tsx`. Similar for when you have the tests for the button component it would be `button.test.tsx`.

Reply:
As an example, our animated_number/index.tsx, would have been `animated_number/AnimatedNumber.tsx` and the test would have been in the same folder by the name of `AnimatedNumber.test.tsx`.  The component name is `AnimatedNumber`.  If I have so many `index.tsx` file open on tabs of vscode, it's just more confusing.

So, if you have a file called ButtonComponentLibrary.tsx, and your component name is ButtonComponentLibrary, then it matches and much easier to open the file.

Just wondering, not trying to get things changed here.

Visual representation of current and files named by component.

![Screenshot 2024-11-19 at 6 57 00 AM](https://github.com/user-attachments/assets/a831777c-6bcd-451a-834c-a4da7b97f05b)
![Screenshot 2024-11-19 at 7 02 07 AM](https://github.com/user-attachments/assets/b13f8d28-ede0-4a01-906d-2b964a9026a3)

Reply:
So... that is an interesting thing we can deal with. One of the problems is that you want to import components as: `path/to/my/component` instead of `path/to/my/component/component`. If you don't have an index file, that is what happens.

In many components, we use the index file to get the state from the database (similar in webapp, where we use it to get the state from redux). But some components don't have a need for that. You may argue that then, we can remove the folder, but having the folder helps organizing other files that may be related.

A solution that we can do is having "empty index files". With some of the migrations I have done in web towards react hooks, we have decided to follow that path. We have an index file that is practically just two lines:
```
import Component from ./component;
export default Component;
```
We can follow that approach here too, but migrating to that approach may be time consuming (and break a bit the blame history).

Reply:
Actually, I'm more of a path/to/my/component/Component.tsx kinda developer/guy.  😺 

So you would have

/src/components/animated_number/AnimatedNumber.tsx
/src/components/animated_number/AnimatedNumber.test.tsx

instead of

/src/components/animated_number/index.tsx

Kinda like react-native Component...

[packages/react-native/Libraries/Components/TextInput/TextInput.js
](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Components/TextInput/TextInput.js)